azure-sink-connector/src/main/java/io/aiven/kafka/connect/azure/AzureBlobSinkTask.java

```diff
@@ -0,0 +1,177 @@
+/*
+ * Copyright 2024 Aiven Oy
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.aiven.kafka.connect.azure;
+
+import java.io.IOException;
+import java.io.OutputStream;
+import java.nio.channels.Channels;
+import java.time.Duration;
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.concurrent.ConcurrentHashMap;
+
+import org.apache.kafka.clients.consumer.OffsetAndMetadata;
+import org.apache.kafka.common.TopicPartition;
+import org.apache.kafka.connect.errors.ConnectException;
+import org.apache.kafka.connect.sink.SinkRecord;
+import org.apache.kafka.connect.sink.SinkTask;
+
+import io.aiven.kafka.connect.common.grouper.RecordGrouper;
+import io.aiven.kafka.connect.common.grouper.RecordGrouperFactory;
+import io.aiven.kafka.connect.common.output.OutputWriter;
+
+import com.azure.core.http.policy.ExponentialBackoffOptions;
+import com.azure.core.http.policy.HttpLogDetailLevel;
+import com.azure.core.http.policy.HttpLogOptions;
+import com.azure.core.http.policy.RetryOptions;
+import com.azure.core.http.policy.UserAgentPolicy;
+import com.azure.storage.blob.BlobClient;
+import com.azure.storage.blob.BlobContainerClient;
+import com.azure.storage.blob.BlobServiceClient;
+import com.azure.storage.blob.BlobServiceClientBuilder;
+import com.azure.storage.blob.specialized.BlockBlobClient;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public final class AzureBlobSinkTask extends SinkTask {
+    private static final Logger LOG = LoggerFactory.getLogger(AzureBlobSinkConnector.class);
+
+    private RecordGrouper recordGrouper;
+    private AzureBlobSinkConfig config;
+    private BlobContainerClient containerClient;
+    private final Map<String, BlockBlobClient> blobClientMap = new ConcurrentHashMap<>();
+
+    // required by Connect
+    public AzureBlobSinkTask() {
+        super();
+    }
+
+    // for testing
+    public AzureBlobSinkTask(final Map<String, String> props, final BlobServiceClient blobServiceClient) {
+        super();
+        Objects.requireNonNull(props, "props cannot be null");
+        Objects.requireNonNull(blobServiceClient, "blobServiceClient cannot be null");
+
+        this.config = new AzureBlobSinkConfig(props);
+        this.containerClient = blobServiceClient.getBlobContainerClient(config.getContainerName());
+        initRest();
```

Comment:
nit: Some of this is non-trivially shared with start(). If the tests don't call start intentionally, then the parts that this constructor doesn't run should be more clear.